assets/app/view/game/part/track.rb

```diff
@@ -29,58 +29,37 @@ def render
           # Array<Array<Path>>
           @routes_paths = @routes.map { |route| route.paths_for(@tile.paths) }
 
-          if @tile.offboards.any?
-            @tile.paths.select(&:offboard).map do |path|
+          tracks = []
+          @tile.paths.select { |path| path.edges.size == 2 }
+          .map { |path| [path, index_for(path)] }
+          .sort_by { |_, index| index || -1 }
+          .map do |path, index|
+            tracks << h(TrackCurvilinearPath, region_use: @region_use, path: path,
+                                              color: value_for_index(index, :color),
+                                              width: value_for(path, :width),
+                                              dash: value_for(path, :dash),)
+          end
+
+          tracks << @tile.paths.map do |path|
```

Comment:
this doesn't make sense you're adding an array of paths to tracks?

Reply:
It's an array of h() objects.

Reply:
`tracks` is also an array of `h()` objects, probably better to `.concat` them, or `.flatten` afterward (though it does all get flattened later, at https://github.com/tobymao/18xx/blob/master/assets/app/view/game/tile.rb#L69)

Reply:
yea, it should be concat

Reply:
OK